driver-core/src/main/java/com/datastax/driver/core/ExecutionInfo.java

```diff
@@ -119,4 +127,8 @@ public ConsistencyLevel getAchievedConsistencyLevel() {
     public QueryTrace getQueryTrace() {
         return trace;
     }
+
+
+
+    public PagingState getPagingState(Statement statement) { return PagingState.wrap(usedPagingState, statement); }
```

Comment:
Actually, this method should not have any parameter, we already know what the statement is. You should build the `PagingState` in `ArrayBackedResultSet` (where you have access to the statement) and pass it to `ExecutionInfo`.